src/read.rs

```diff
@@ -128,9 +128,15 @@ use crate::read::lzma::LzmaDecoder;
 #[cfg(feature = "xz")]
 use crate::read::xz::XzDecoder;
 use crate::result::ZipError::{InvalidArchive, InvalidPassword, UnsupportedArchive};
-use crate::spec::is_dir;
 use crate::types::ffi::S_IFLNK;
 use crate::unstable::{path_to_string, LittleEndianReadExt};
+
+use crate::crc32::non_crypto::Crc32Reader as NewCrc32Reader;
+use crate::unstable::read::{
+    construct_decompressing_reader, find_entry_content_range, CryptoEntryReader, CryptoVariant,
+};
+pub use crate::unstable::read::{ArchiveEntry, ZipEntry};
```

Comment:
This doesn't need to be public.
```suggestion
pub(crate) use crate::unstable::read::{ArchiveEntry, ZipEntry};
```